### PR TITLE
Added Accept Route 

### PR DESCRIPTION
**Pull request by tanushgolwala**
The Accept Route for Booking Route has been created

### CODE REVIEW

**Comment by tanushgolwala** [comment]
#19

File: models/UserModel.go [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-	YOB               uint   `json:"yob" valid:"range(1900|2100)~Year of birth must be between 1900 and 2100"`
-	Rides             []Ride `gorm:"foreignKey:HostUserID" json:"past_rides"`
+	Name              string        `gorm:"type:varchar(100);not null;" json:"name" valid:"required~Name is required,matches(^[a-zA-Z ]+$)~Name must be alphabetic"`
+	Email             string        `gorm:"type:varchar(100);not null;uniqueIndex" json:"email" valid:"required~Email is required,email~Email is not valid"`
+	ProfilePictureURL string        `gorm:"type:text" json:"profile_picture_url" valid:"url~URL is not valid"`
+	ContactNumber     string        `gorm:"type:varchar(20);not null" json:"contact_number" valid:"required~Contact number is required,numeric~Contact number must be numeric"`
+	Gender            string        `gorm:"type:varchar(10)" json:"gender" valid:"in(male|female|other)~Gender must be male female or other"`
+	YOB               uint          `json:"yob" valid:"range(1900|2100)~Year of birth must be between 1900 and 2100"`
+	RidesID           pq.Int64Array `gorm:"type:integer[]" json:"ride_ids"`
```
Comment (RohanKhatua) [inline models/UserModel.go:18]: Why does the user need to have an array of ride IDs in the user table, why can they just not be referenced from the user id fkey relation present in the rides table?

File: routes/bookings/AcceptRoute.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/gofiber/fiber/v2"
+)
+
+func AcceptRoute(c *fiber.Ctx) error {
+	bookingID := c.Params("bookingID")
+
+	var booking models.Booking
+
+	if err := database.Database.Db.First(&booking, bookingID).Error; err != nil {
```
Comment (RohanKhatua) [inline routes/bookings/AcceptRoute.go:17]: Multiple DB operations part of a single API call should be part of a transaction.

File: routes/bookings/AcceptRoute.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		return c.Status(400).SendString("No available seats for this ride")
+	}
+
+	// Increment the booked seats count
+	if err := database.Database.Db.Model(&ride).Update("booked_seats", ride.BookedSeats+1).Error; err != nil {
+		log.Printf("Error updating booked seats for ride: %v\n", err)
+		return c.Status(500).SendString("Error updating booked seats for ride")
+	}
+
+	var User models.User
```
Comment (RohanKhatua) [inline routes/bookings/AcceptRoute.go:45]: Again, this does not need to be done, if the relation is accessed via the fkey constraint

**Review by RohanKhatua — CHANGES_REQUESTED** [review]
See review comments

File: routes/users/getUserRides.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"carpool-backend/database"
+	"carpool-backend/models"
+	"log"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+type RideResponse struct {
+	ID            uint   `json:"ID"`
+	CreatedAt     string `json:"CreatedAt"`
```
Comment (RohanKhatua) [inline routes/users/getUserRides.go:13]: Do not return CreatedAt and UpdatedAt in api response - unnecessary

File: routes/users/getUserRides.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	for i, booking := range bookings {
+		rideIDs[i] = booking.RideID
+	}
+
+	// Retrieve rides associated with the extracted ride IDs
+	var rides []models.Ride
+	if err := database.Database.Db.Where("id IN (?)", rideIDs).Find(&rides).Error; err != nil {
+		log.Printf("Error finding rides: %v\n", err)
+		return c.Status(500).SendString("Error finding rides")
+	}
```
Comment (RohanKhatua) [inline routes/users/getUserRides.go:47]: Just do a join.

File: routes/bookings/AcceptRoute.go (hunk 2 above)
Comment (RohanKhatua) [reply routes/bookings/AcceptRoute.go:17]: don't see a tx yet

**Review by RohanKhatua — CHANGES_REQUESTED** [review]

**Review by RohanKhatua — APPROVED** [review]
LGTM

**Pull request merged**